### ISSUE TITLE
QSelect choices automatically show up on tab?

### ISSUE THREAD

**Issue by knoxvilledatabase** [elimintz/justpy#75]
Is there anyway to have the choices popup automatically when a user tabs to a QSelect field? 

For example, I have a form with 3 text input fields and a QSelect field. As the user is filling out a fields then tabbing to the next field, when they get to the QSelect they have to do an additional click inside the field to get the choices to popup. I'm curious, can the choices pop up from a tab? 

Thank you!

**Comment by elimintz** [comment]
Yes, this should be doable. You would need to add an event handler to the focus event that runs the method to open the popup.

However, when I try to create an example I found a bug in Quasar. While QSelect responds to focus event it puts foucsin as the event type and JustPy cannot find the handler. Let me see what I can do. But it may be the case that this can only be fixed in the next version.

**Comment by elimintz** [comment]
Sorry, I had to make some changes in the JavaScript to make this work so it will be available only in next version.

This is the reference program:

```
import justpy as jp

async def focus_in(self, msg):
    await self.run_method('showPopup()', msg.websocket)
    return True

def select_test():
    wp = jp.QuasarPage()
    d = jp.Div(classes="q-pa-md", style="max-width: 300px", a=wp)
    d1 = jp.Div(classes="q-gutter-md", a=d)

    for i in range(5):
        s1 = jp.QSelect(label='Label', a=d1, color='orange', standout=True)
        print(s1.allowed_events)
        s1.options = ['Google', 'Facebook', 'Twitter', 'Apple', 'Oracle']
        s1.on('focus', focus_in)
    return wp


jp.justpy(select_test)
```

**Comment by knoxvilledatabase** [comment]
Should this work in 0.1.1?

Thank you!

**Comment by elimintz** [comment]
Yes it should. Are you having a problem?

**Comment by knoxvilledatabase** [comment]
Let me try again, it must be something I am doing wrong.

**Comment by knoxvilledatabase** [comment]
So, it seams that when I have focus_in enabled on the QInput the tab works great but when I click on the field I have to click again for the options to pop up. Would it be possible to have the options pop up by either tab or click without requiring a second click?

**Comment by elimintz** [comment]
The following works (adding a 200 ms sleep in focus_in):

```python
import justpy as jp
import asyncio

async def focus_in(self, msg):
    await asyncio.sleep(0.2)
    await self.run_method('showPopup()', msg.websocket)
    return True

def select_test():
    wp = jp.QuasarPage()
    d = jp.Div(classes="q-pa-md", style="max-width: 300px", a=wp)
    d1 = jp.Div(classes="q-gutter-md", a=d)

    for i in range(5):
        s1 = jp.QSelect(label='Label', a=d1, color='orange', standout=True)
        print(s1.allowed_events)
        s1.options = ['Google', 'Facebook', 'Twitter', 'Apple', 'Oracle']
        s1.on('focus', focus_in)
    return wp


jp.justpy(select_test)
```

However, it is not a great solution since it depends on the timing of the sleep in focus_in. If the sleep time is too short, depending on the latency, it does not work. 0.2 seconds works fine with 40 ms latency. If the sleep time is larger, the users may see a lag in the select opening. Experiment with it and see what works for your users.

For the next version I will try to figure out a more robust solution. The QSelect component supports some events that could be useful in this case.

**Comment by knoxvilledatabase** [comment]
Worked like a charm, big thanks!
